tests/test_models/test_mutables/test_gumbelchoiceroute.py

```diff
@@ -32,8 +33,10 @@ def test_forward_arch_param(self):
 
         # test with_arch_param = True
         GumbelChoiceRoute = MODELS.build(gumbel_choice_route_cfg)
+        mutator = DiffModuleMutator()
+        mutator.prepare_from_supernet(GumbelChoiceRoute)
 
-        arch_param = GumbelChoiceRoute.build_arch_param()
+        arch_param = mutator.build_arch_param(GumbelChoiceRoute.num_choices)
```

Comment:
Use user-defined instead  of `mutator`  to construct `arch_param`, as we hope the mutable could work(build and forward) independently in our design.